### PR TITLE
CORE-2915 - Mgm sync service

### PR DESCRIPTION
**Pull request by nikinagy**
https://r3-cev.atlassian.net/browse/CORE-2915

### CODE REVIEW

File: data/avro-schema/src/main/resources/avro/net/corda/data/membership/command/synchronisation/mgm/ProcessSyncRequest.avsc [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+  "type": "record",
+  "name": "ProcessSyncRequest",
+  "namespace": "net.corda.data.membership.command.synchronisation.mgm",
+  "doc": "Command issued for an mgm to process the sync request sent by a member.",
+  "fields": [
+    {
+      "name": "mgm",
```
Comment (yift-r3) [inline data/avro-schema/src/main/resources/avro/net/corda/data/membership/command/synchronisation/mgm/ProcessSyncRequest.avsc]: Isn't the `mgm` the same as the destination of the P2P message?

File: data/avro-schema/src/main/resources/avro/net/corda/data/membership/command/synchronisation/mgm/ProcessSyncRequest.avsc [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      "doc": "Holding identity of the MGM.",
+      "type": "net.corda.data.identity.HoldingIdentity"
+    },
+    {
+      "name": "requester",
```
Comment (yift-r3) [inline data/avro-schema/src/main/resources/avro/net/corda/data/membership/command/synchronisation/mgm/ProcessSyncRequest.avsc]: Isn't the `requester` the same as the source?

File: data/avro-schema/src/main/resources/avro/net/corda/data/membership/command/synchronisation/mgm/ProcessSyncRequest.avsc (hunk 1 above)
Comment (nikinagy) [reply data/avro-schema/src/main/resources/avro/net/corda/data/membership/command/synchronisation/mgm/ProcessSyncRequest.avsc]: Yes, it is, but we are not passing the whole authenticated messages into the sync service, hence we won't have the header information.

File: data/avro-schema/src/main/resources/avro/net/corda/data/membership/command/synchronisation/mgm/ProcessSyncRequest.avsc (hunk 2 above)
Comment (nikinagy) [reply data/avro-schema/src/main/resources/avro/net/corda/data/membership/command/synchronisation/mgm/ProcessSyncRequest.avsc]: See my other comment.

**Review by charlieR3 — DISMISSED** [review]

**Review by charlieR3 — APPROVED** [review]

**Pull request merged**